### DISCUSSION TITLE
Option to require password for updateUser()

### DISCUSSION THREAD

**Discussion by janglad** [supabase/supabase#18820] [Feature Requests]
### Problem
I'm relatively new to web dev but one thing I've really internalized is "never trust anything on/from the client". Using Supabase it's to be expected that you're exposing the Supabase client on the client side using the anon/public key. In other words, there always is a possibility that a malicious user has access to the Supabase client.

This combined with RLS policies etc can make for a great experience, but I am somewhat worried about the implications of this on impactful functions like `updateUser()`. As of right now, there doesn't seem to be any way to require the current user password to be included and checked before updating data that could lock a user out of their account. The closest option is requiring a fresh session, which still has a margin of 24 hours. 

### What I'd want
Make an option in the Supabase dashboard to require the current user password to be present and checked when calling `updateUser()`, or alternatively provide an option to require a verification code/nonce to be sent when a user updates their account password/email, even if the session is "fresh".

### Why I want it 
Prevent a malicious actor from hijacking an account purely based on session.

**Comment by Donnerstagnacht** [comment]
I would double this. I think it is a basic feature which many apps out there use/require. It would avoid this complex and developer unfriendly workaround: https://github.com/orgs/supabase/discussions/4042
